### PR TITLE
Improved the Describe > One Variable > Summarise dialog

### LINKED ISSUE #7081: Improve the Describe > One Variable > Summarise dialogue and possibly 2-variable and 3-variable?
I use the data from the R-Instat review as an example:

I use the one variable summarise and wanted to get the sum (as in the review) and also the number of occasions with more than zero.
Here is the output:
```

# Code generated by the dialog, One Variable Summarise
data_book$summary(data_name="Scores", na.rm=TRUE, count_test="'>'", count_value=0, j=1, return_output=TRUE, summaries=c("count_calc", "summary_count_non_missing", "summary_count", "summary_sum"), columns_to_summarise=c("BlueSky","Deducer","JASP","jamovi","RAF","Rattle","Rcmdr","R_Instat","RKWard"))

                  BlueSky Deducer JASP jamovi RAF Rattle Rcmdr R_Instat RKWard
count                  43      43   43     43  43     43    43       43     43
count-non-missing      43      37   39     40  37     38    37       43     35
sum                   359     102  224    235  59    110   686      290    100
count-calc             NA      NA   NA     NA  NA     NA    NA       NA     NA

```
The main changes I would like are to the layout.  But here there is an oddity in the calculation.  What am I doing wrong?  The omit missing values option is ticked?  The count-calc is count values > 0, i.e. the number of values of 1 or more.  
This points to the urgency of doing something here.  But now I remember that we had the same issue with our Describe > Specific > Summary Tables dialogue.  And we solved the problem there!

Aha I have the solution in general.  One thing I would like is to be able to control the orientation, i.e. choose whether it is the variables or the summaries that go down.  I am sure there will be more.  But this is (almost) a special case of the Summary tables dialogue.  I say almost, because that dialogue is limited to numeric variables to summarise, while here we can summarise any type - though dates become numeric!  Anyway we can summarise factors and ordered factors include max and min.  (Do we want an option to do ordinary summarise on numeric levels of a factor - I guess not???)
We might give the results separately for the factors and the variates?

And I suggest these special cases will also be ok for the 2-variable and 3-variable summarise as well!  We will have a limited set of options - that's to keep it simple.  If you want full generality then you go to the main dialogues under the Specific menu.

I assume this is mainly work for @Ivanluv, with possible guidance from @lilyclements.  He will, I hope, be able to sub-contract some of the work to one of the interns?

### PR DESCRIPTION
**Pull request by Ivanluv**
fixes #7081 and replaces PR #7123

@africanmathsinitiative/developers this is ready for review

### CODE REVIEW

**Comment by lloyddewit** [comment]
@HawardKetoyoMsatsi Please could you review?
@rdstern Please could you test?
Thanks!

**Review by rdstern — CHANGES_REQUESTED** [review]
@Ivanluv and @lilyclements this is excellent now.  All over except the shouting!  I used the usual survey data and could get all I wanted in terms of the summary statistics.
Just some cosmetic things remain. I think:
a) Is the terminology the wrong way round?  The default is Display summaries as rows.  I think you mean columns?  When I make both summaries and variables as rows, then I only get one row and many columns.  So maybe all you need to change is the label.
b) When I untick both, there is just a single column.  And there is no labelling, just the numbers.
c) There is the margin All, over which I have no control.  In my examples it just gives the total count.  I'm not sure what that means and am not sure I want  the margin here?

**Comment by Ivanluv** [comment]
@rdstern could you have a look again?

File: instat/dlgOneVariableSummarise.vb [hunk 1]
```diff
@@ -162,11 +210,48 @@ Public Class dlgOneVariableSummarise
         strDefaultColumns = Nothing
     End Sub
 
-    Private Sub Controls_ControlContentsChanged(ucrChangedControl As ucrCore) Handles ucrReceiverOneVarSummarise.ControlContentsChanged, ucrNudMaxSum.ControlContentsChanged, ucrPnlSummaries.ControlValueChanged
+    Private Sub Controls_ControlContentsChanged(ucrChangedControl As ucrCore) Handles ucrReceiverOneVarSummarise.ControlContentsChanged, ucrNudMaxSum.ControlContentsChanged
```
Comment (HawardKetoyoMsatsi) [inline instat/dlgOneVariableSummarise.vb:226]: I guess you need to listen to events happening in ucrReceiverOneVarSummarise and clsSummariesList

File: instat/dlgOneVariableSummarise.vb [hunk 2]
```diff
@@ -162,11 +210,48 @@ Public Class dlgOneVariableSummarise
...
         TestOKEnabled()
     End Sub
 
     Private Sub ucrPnlSummaries_ControlValueChanged(ucrChangedControl As ucrCore) Handles ucrPnlSummaries.ControlValueChanged
-        ChangeBaseFunction()
+        If rdoCustomised.Checked Then
+            clsDummyFunction.AddParameter("checked_radio", "customised", iPosition:=2)
+            ucrBase.clsRsyntax.SetBaseROperator(clsMmtableOperator)
+            cmdSummaries.Visible = True
+        ElseIf rdoDefault.Checked Then
```
Comment (HawardKetoyoMsatsi) [inline instat/dlgOneVariableSummarise.vb:235]: could this just be an Else instead of ElseIf

File: instat/dlgOneVariableSummarise.vb [hunk 3]
```diff
@@ -162,11 +210,48 @@ Public Class dlgOneVariableSummarise
...
+    Private Sub Controls_ControlContentsChanged(ucrChangedControl As ucrCore) Handles ucrReceiverOneVarSummarise.ControlContentsChanged, ucrNudMaxSum.ControlContentsChanged
         TestOKEnabled()
     End Sub
 
     Private Sub ucrPnlSummaries_ControlValueChanged(ucrChangedControl As ucrCore) Handles ucrPnlSummaries.ControlValueChanged
-        ChangeBaseFunction()
+        If rdoCustomised.Checked Then
+            clsDummyFunction.AddParameter("checked_radio", "customised", iPosition:=2)
+            ucrBase.clsRsyntax.SetBaseROperator(clsMmtableOperator)
+            cmdSummaries.Visible = True
```
Comment (HawardKetoyoMsatsi) [inline instat/dlgOneVariableSummarise.vb:234]: Could cmdSummaries be set as setLinkedDisplayControl

File: instat/dlgOneVariableSummarise.vb [hunk 4]
```diff
@@ -162,11 +210,48 @@ Public Class dlgOneVariableSummarise
...
+        ElseIf rdoDefault.Checked Then
+            clsDummyFunction.AddParameter("checked_radio", "defaults", iPosition:=2)
+            ucrBase.clsRsyntax.SetBaseRFunction(clsSummaryFunction)
+            cmdSummaries.Visible = False
```
Comment (HawardKetoyoMsatsi) [inline instat/dlgOneVariableSummarise.vb:238]: Same here

**Review by HawardKetoyoMsatsi — DISMISSED** [review]
@Ivan I requested for a few changes, please take a look.

File: instat/dlgOneVariableSummarise.vb [hunk 5]
```diff
@@ -20,9 +20,16 @@ Imports instat.Translations
...
+    Private clsSummaryFunction, clsSummariesList, clsMmtableFunction,
+        clsConcFunction, clsSummaryTableFunction, clsHeaderLeftTopVariableFunction,
+                clsHeaderLeftTopSummaryFunction, clsHeaderTopLeftVariableFunction,
+                clsHeaderTopLeftSummaryFunction, clsDummyFunction As New RFunction
+    Private clsMmtableOperator As ROperator
     Private bResetSubdialog As Boolean = False
     Public strDefaultDataFrame As String = ""
+
+
+
```
[the remark this answers is not in the document]
Comment (Ivanluv) [reply instat/dlgOneVariableSummarise.vb:33]: Done

File: instat/dlgOneVariableSummarise.vb (hunk 1 above)
Comment (Ivanluv) [reply instat/dlgOneVariableSummarise.vb:226]: Its already done

File: instat/dlgOneVariableSummarise.vb (hunk 2 above)
Comment (Ivanluv) [reply instat/dlgOneVariableSummarise.vb:235]: Done

File: instat/dlgOneVariableSummarise.vb (hunk 3 above)
Comment (Ivanluv) [reply instat/dlgOneVariableSummarise.vb:234]: That is not possible when the `cmdSummaries`   radio buttons within the panel

File: instat/dlgOneVariableSummarise.vb (hunk 4 above)
Comment (Ivanluv) [reply instat/dlgOneVariableSummarise.vb:238]: That's not possible

**Review by rdstern — APPROVED** [review]
@Ivanluv and @lilyclements  - great!

**Comment by lloyddewit** [comment]
> @HawardKetoyoMsatsi ?

@HawardKetoyoMsatsi Thank you for your review. I went through the comments/responses and set the comments to resolved. If I misunderstood anything, then let me know.
Thanks

**Review by lloyddewit — APPROVED** [review]
@Ivanluv Thank you! looks good

**Comment by lloyddewit** [comment]
@Ivanluv does this PR fully resolve #7081? Or just partially?

**Comment by Ivanluv** [comment]
@lloyddewit  this fully fixes #7081

**Pull request merged**